### PR TITLE
Adds count class level permission

### PR DESCRIPTION
**Pull request by flovilmart**
Partial resolution for #3813

### CODE REVIEW

File: spec/CloudCode.spec.js [hunk 1]
```diff
@@ -1470,4 +1470,36 @@ describe('afterFind hooks', () => {
...
+    spyOn(hook, 'method').and.callThrough();
+    Parse.Cloud.beforeFind('Stuff', hook.method);
+    new Parse.Query('Stuff').count().then((count) => {
+      expect(count).toBe(0);
+      expect(hook.method).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it('should report count if passed', (done) => {
```
Comment (acinader) [inline spec/CloudCode.spec.js]: the test description is the same as above.  maybe should be 'should report find if passed', but i'm confused what you're testing and why.  what does it mean "to report" in this context?

i just see that the result is zero as expected and your hook is called for both find and count.

File: spec/Schema.spec.js [hunk 2]
```diff
@@ -178,6 +178,38 @@ describe('SchemaController', () => {
...
+    var obj;
+    return config.database.loadSchema()
+      // Create a valid class
+      .then(schema => schema.validateObject('Stuff', {foo: 'bar'}))
+      .then(schema => {
+        var count = {};
+        return schema.setPermissions('Stuff', {
+          'create': {'*': true},
+          'find': {'*': true},
+          'count': count
```
Comment (acinader) [inline spec/Schema.spec.js]: i'm assuming that `count = {} ` is roughly equivelent to `count = { '*': false }` ?

File: spec/Schema.spec.js [hunk 3]
```diff
@@ -178,6 +178,38 @@ describe('SchemaController', () => {
...
 
+  it('class-level permissions test count', (done) => {
+    var obj;
+    return config.database.loadSchema()
+      // Create a valid class
+      .then(schema => schema.validateObject('Stuff', {foo: 'bar'}))
+      .then(schema => {
+        var count = {};
+        return schema.setPermissions('Stuff', {
+          'create': {'*': true},
```
Comment (acinader) [inline spec/Schema.spec.js]: no quotes around object members

File: src/triggers.js [hunk 4]
```diff
@@ -155,11 +155,12 @@ export function getRequestObject(triggerType, auth, parseObject, originalParseOb
   return request;
 }
 
-export function getRequestQueryObject(triggerType, auth, query, config) {
+export function getRequestQueryObject(triggerType, auth, query, count, config) {
   var request = {
     triggerName: triggerType,
     query: query,
     master: false,
+    count: count,
```
Comment (acinader) [inline src/triggers.js]: fyi, you *could* just have `count` instead of `count: count`.

Comment (flovilmart) [inline src/triggers.js]: That was to match the style, I'll update :)

File: spec/Schema.spec.js (hunk 2 above)
Comment (flovilmart) [inline spec/Schema.spec.js]: yeah but *: false is not a valid CLP, {} means lockdown, no one can read / write

File: spec/Schema.spec.js (hunk 3 above)
Comment (flovilmart) [inline spec/Schema.spec.js]: That was to match the style of the rest of the file, we'll update those at a later time (i.e. test linting)

**Comment by flovilmart** [comment]
@acinader pushed some nits

**Pull request merged**